lib/core/settle.js

```diff
@@ -11,7 +11,8 @@ var createError = require('./createError');
  */
 module.exports = function settle(resolve, reject, response) {
   var validateStatus = response.config.validateStatus;
-  if (!validateStatus || validateStatus(response.status)) {
+  var responseUrl = response.config.url;
+  if (!validateStatus || validateStatus(response.status) || (!validateStatus && responseUrl.indexOf('file:') === 0)) {
```

Comment:
@yasuf @Alanscut Before merging, do you know why 0.18 didn't have this problem, even without this check? I think we should find the root reason.

Reply:
@chinesedfan  request that using file: protocol, most browsers will return status as 0 even though it's a successful request. According to the code of v0.18.0, we can get a correct response. See #1689.

v0.18.0 
```javascript
if (!response.status || !validateStatus || validateStatus(response.status)) {
  resolve(response);
} else {
  reject(createError(
    'Request failed with status code ' + response.status,
    response.config,
    null,
    response.request,
    response
  ));
}
```

v0.19.0

```javascript
if (!validateStatus || validateStatus(response.status)) {
  resolve(response);
} else {
  reject(createError(
    'Request failed with status code ' + response.status,
    response.config,
    null,
    response.request,
    response
  ));
}
```

Reply:
So, how about reverting those lines without the extra protocol check?

```
  // Note: status is not exposed by XDomainRequest and is 0 for file://
  if (!response.status
```

Reply:
Remove the protocol check,   personally think that there will be no errors. I will resubmit the code, thanks.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Reply:
@Alanscut Thanks for your fix. Would you mind adding a test case for it? To avoid someone removes it by accident again.

Reply:
@chinesedfan Unit tests have been added, thanks.